src/styles/blocks/box.scss

```diff
@@ -0,0 +1,24 @@
+.box {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 300px;
+  min-width: 300px;
+
+  color: #fff;
+  font-size: 100px;
+}
+
+$box-color:
+  "1" #f00,
+  "2" #c00,
+  "3" #900,
+  "4" #600,
+  "5" #300,
+  "6" #000;
+
+@each $number, $color in $box-color {
+  .box--#{$number} {
+    background-color: #{$color};
+  }
+}
```

Comment:
I think @each shoud be inside class .box. You can try use nth:child, and remove mod .box--number